### PR TITLE
feat(docs): `Builder.storeMaybeRef`, `parseStdAddress` and `parseVarAddress`

### LINKED ISSUE #900: Document `Builder.storeMaybeRef`, `parseStdAddress` and `parseVarAddress` stdlib functions
https://github.com/tact-lang/tact/pull/793

### PR DESCRIPTION
**Pull request by novusnota**
And documentation comments to `stdlib/`, plus descriptions of `StdAddress` and `VarAddress`

## Issue

Closes #900 and towards #573.

## Checklist

- [x] I have updated CHANGELOG.md
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

File: docs/src/content/docs/ref/core-advanced.mdx [hunk 1]
```diff
@@ -251,6 +253,80 @@ nativeReserve(ton("0.1"), ReserveExact | ReserveBounceIfActionFail);
 //            amount of nanoToncoins to reserve
 ```
 
+## parseStdAddress
```
Comment (anton-trunov) [inline docs/src/content/docs/ref/core-advanced.mdx:256]: We should mention the connection between `newAddress` and `parseStdAddress`:

```tact
let stdAddr = parseStdAddress(addr.asSlice());
newAddress(stdAddr.workchain, stdAddr.address) == addr;
```

File: docs/src/content/docs/ref/core-advanced.mdx [hunk 2]
```diff
@@ -251,6 +253,80 @@ nativeReserve(ton("0.1"), ReserveExact | ReserveBounceIfActionFail);
...
+Attempts to pass a [`Slice{:tact}`][slice] with layout different from the `StdAddress{:tact}` or to load more data than a given [`Slice{:tact}`][slice] contains throw an exception with [exit code 9](/book/exit-codes#9): `Cell underflow`.
+
+Usage example:
+
+```tact
+let addr = address("EQDtFpEwcFAEcRe5mLVh2N6C0x-_hJEM7W61_JLnSF74p4q2");
+let parsedAddr = parseStdAddress(addr.asSlice());
+
+parsedAddr.workchain; // 0
+parsedAddr.address;   // 107...287
```
Comment (anton-trunov) [inline docs/src/content/docs/ref/core-advanced.mdx]: what does `107...287` mean?

File: docs/src/content/docs/ref/core-cells.mdx [hunk 3]
```diff
@@ -286,6 +290,31 @@ let b: Builder = beginCell();
...
+
+As a single [`Cell{:tact}`][cell] can store up to $4$ references, attempts to store more throw an exception with [exit code 8](/book/exit-codes#8): `Cell overflow`.
+
+Usage example:
+
+```tact
+let b: Builder = beginCell();
+let fizz: Builder = b
+    .storeMaybeRef(emptyCell()) // 1, then empty cell
+    .storeMaybeRef(null);       // 0
```
Comment (anton-trunov) [inline docs/src/content/docs/ref/core-cells.mdx]: what do `1` and `0` mean here?

**Review by anton-trunov — CHANGES_REQUESTED** [review]

File: docs/src/content/docs/ref/core-advanced.mdx (hunk 2 above)
Comment (novusnota) [reply docs/src/content/docs/ref/core-advanced.mdx]: Address integer, but truncated in-between with `...` not to consume too much horizontal space

File: docs/src/content/docs/ref/core-cells.mdx (hunk 3 above)
Comment (novusnota) [reply docs/src/content/docs/ref/core-cells.mdx]: Single `1` and `0` bits

Comment (anton-trunov) [reply docs/src/content/docs/ref/core-cells.mdx]: this looks confusing to me, I'd either add a TL-B description or removed the comments altogether

File: docs/src/content/docs/ref/core-advanced.mdx (hunk 1 above)
Comment (novusnota) [reply docs/src/content/docs/ref/core-advanced.mdx:256]: Added a note by the end of the usage example

File: docs/src/content/docs/ref/core-advanced.mdx (hunk 2 above)
Comment (novusnota) [reply docs/src/content/docs/ref/core-advanced.mdx]: Explained ... better

File: docs/src/content/docs/ref/core-cells.mdx (hunk 3 above)
Comment (novusnota) [reply docs/src/content/docs/ref/core-cells.mdx]: Elaborated :)

**Comment by novusnota** [comment]
@Gusarich please, take a look!

**Review by Gusarich — APPROVED** [review]
LGTM! Sorry for the delay

**Pull request merged**